### PR TITLE
Ep 1467 code documentation

### PR DESCRIPTION
**Pull request by marcghelas**

### CODE REVIEW

File: judokit-android/src/main/java/com/judokit/android/Judo.kt [hunk 1]
```diff
@@ -71,59 +78,110 @@ class Judo internal constructor(
...
+         * @param address An object containing information about the card address.
+         * If set, will be sent to the Judo backend as part of the card details
+         * @see Address
+         */
         fun setAddress(address: Address?) =
             apply { this.address = address }
 
+        /**
+         * Sets Pay by Bank app configuration.
+         * @param pbbaConfiguration An instance of JPPbbaConfiguration required for PbBA
```
Comment (mpetrenco) [inline judokit-android/src/main/java/com/judokit/android/Judo.kt:171]: Isn't the JP prefix iOS only?

File: judokit-android/src/main/java/com/judokit/android/model/PaymentWidgetType.kt [hunk 2]
```diff
@@ -4,18 +4,71 @@ import android.os.Parcelable
...
+     */
     CARD_PAYMENT,
+
+    /**
+     * Starts a pre-auth card payment flow.
+     */
     PRE_AUTH,
+
+    /**
+     * Starts a register card flow for making future tokenized payments
```
Comment (mpetrenco) [inline judokit-android/src/main/java/com/judokit/android/model/PaymentWidgetType.kt:25]: Register card differs from Save card in respect to the fact that Register Card hardcodes the amount to 1.01. This amount is taken from my account when I call this method. Might need to look at use cases when Register Card is preferable to Save Card for card tokenization.

File: judokit-android/src/main/java/com/judokit/android/model/PaymentMethod.kt [hunk 3]
```diff
@@ -5,11 +5,33 @@ import com.judokit.android.R
 import com.judokit.android.ui.paymentmethods.components.PaymentButtonType
 import kotlinx.android.parcel.Parcelize
 
+/**
+ * A set of available payment methods to use in payment methods screen.
+ * If no payment method is specified, then only card payment will be available.
```
Comment (mpetrenco) [inline judokit-android/src/main/java/com/judokit/android/model/PaymentMethod.kt:10]: Weird stuff. On iOS, if no payment method is set, then all are available by default.

File: judokit-android/src/main/java/com/judokit/android/model/PBBAConfiguration.kt [hunk 4]
```diff
@@ -4,6 +4,10 @@ import android.net.Uri
 import android.os.Parcelable
 import kotlinx.android.parcel.Parcelize
 
+/**
+ * A configuration class responsible for setting additional parameters for
+ * pay by Bank app payment method.
```
Comment (mpetrenco) [inline judokit-android/src/main/java/com/judokit/android/model/PBBAConfiguration.kt:9]: pay -> Pay

File: judokit-android/src/main/java/com/judokit/android/model/CardNetwork.kt [hunk 5]
```diff
@@ -62,19 +75,35 @@ enum class CardNetwork : Parcelable {
...
 val CardNetwork.cardNumberMask: String
     get() = when (this) {
         CardNetwork.AMEX -> "#### ###### #####"
         CardNetwork.DINERS_CLUB -> "#### ###### ####"
         else -> DEFAULT_CARD_NUMBER_MASK
     }
 
+/**
+ * Extension property that returns the mask of the security code based on the card network.
+ * @return Mask of the secuirty code.
```
Comment (mpetrenco) [inline judokit-android/src/main/java/com/judokit/android/model/CardNetwork.kt:91]: secuirty

**Review by mpetrenco — APPROVED** [review]
That's actually a really nice documentation 👌

File: judokit-android/src/main/java/com/judokit/android/Judo.kt (hunk 1 above)
Comment (marcghelas) [reply judokit-android/src/main/java/com/judokit/android/Judo.kt:171]: yes, I will fix that

File: judokit-android/src/main/java/com/judokit/android/model/PaymentMethod.kt (hunk 3 above)
Comment (marcghelas) [reply judokit-android/src/main/java/com/judokit/android/model/PaymentMethod.kt:10]: Strange. Maybe add this as tech debt and check with Jumaane how he wants

**Review by stpopa — APPROVED** [review]
👍

**Pull request merged**